src/main/java/Parser.java

```diff
@@ -0,0 +1,234 @@
+        }
+        String[] deadlineParsed = rest.split("/");
+        if (deadlineParsed.length == 1) {
+            throw new DukeException("Prefix the keyword 'by' with a forward slash!");
+        }
+        String deadlineName = deadlineParsed[0].trim();
+        String[] byParsed = deadlineParsed[1].split(" ", 2);
+        if (byParsed.length == 1) {
+            throw new DukeException("Deadline due time cannot be empty!");
+        }
+        String by = byParsed[1];
+        Deadline deadline = new Deadline(deadlineName, by);
+        taskList.addTask(deadline);
+        ui.print("Got it. I've added this task:\n" + deadline + "\n" + taskList.taskSizeString());
+    }
+
+    /**
+     * Handles the creation of an Event task.
+     *
+     * @param rest The remaining string after the key command "event".
+     * @throws DukeException Thrown when invalid format used.
+     */
+    private void handleEvent(String rest) throws DukeException {
+        if (rest == null) {
+            throw new DukeException("Description of an event cannot be empty!");
+        }
+        String[] eventParsed = rest.split("/");
+        if (eventParsed.length == 1) {
+            throw new DukeException("Prefix the keyword 'at' with a forward slash!");
+        }
+        String eventName = eventParsed[0].trim();
+        String[] atParsed = eventParsed[1].split(" ", 2);
+        if (atParsed.length == 1) {
+            throw new DukeException("Event time cannot be empty!");
+        }
+        String at = atParsed[1];
+        Event event = new Event(eventName, at);
+        taskList.addTask(event);
+        ui.print("Got it. I've added this task:\n" + event + "\n" + taskList.taskSizeString());
+    }
+
+    /**
+     * Handles the deletion of a task.
+     *
+     * @param rest The remaining string after the key command "delete", which is index of task.
+     * @throws DukeException Thrown when invalid format used.
+     */
+    private void handleDelete(String rest) throws DukeException {
+        if (rest == null) {
+            throw new DukeException("Specify a task!");
+        }
+        int deleteIndex;
+        try {
+            deleteIndex = Integer.parseInt(rest) - 1;
+        } catch (NumberFormatException ex) {
+            throw new DukeException("Specify a valid task number!");
+        }
+        if (deleteIndex < 0 || deleteIndex >= taskList.size()) {
+            throw new DukeException("No such task!");
+        }
+        Task taskToDelete = taskList.get(deleteIndex);
+        taskList.deleteTask(deleteIndex);
+        ui.print("Noted. I've removed this task:\n" + taskToDelete + "\n" + taskList.taskSizeString());
+    }
+
+    /**
+     * Handles the "find" command.
+     *
+     * @param rest Remaining string after the "find" command, which is query term.
+     * @throws DukeException Thrown when no search term is given.
+     */
+    private void handleFind(String rest) throws DukeException {
```

Comment:
Perhaps a argument name like <code>remainderString</code> would fit better to tell a reader immediately what the argument is